server/models/role.js

```diff
@@ -0,0 +1,28 @@
+export default (sequelize, DataTypes) => {
+  const Role = sequelize.define('Role', {
+    id: {
+      type: DataTypes.UUID,
+      primaryKey: true,
+      defaultValue: DataTypes.UUIDV4
+    },
+    type: {
+      type: DataTypes.STRING,
+      unique: {
+        message: 'the role type must be unique'
+      },
+      validate: {
+        isAlphanumeric: {
+          msg: 'The role title must be alphanumeric'
+        }
+      }
+    }
+  });
+
+  Role.associate = (models) => {
+    Role.hasMany(models.User, {
+      foreignKey: 'role'
```

Comment:
Shouldn't foreign key be roleId instead of role

Reply:
Hi @JohnMadakin. @theghostyced here. So admittedly this was misleading, the reason this is role and not roleId is because using roleId would indicate we are storing a uuid into the table but what is being stored is a string. This is done so as to limit the amount of query sent to the role DB to verify if the uuid is for either a superAdmin or a regular user

Reply:
ok